CHANGELOG.md

```diff
@@ -3,6 +3,15 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+## [Unreleased]
+
+### API changes
+
+- Removed callback argument from `HDDlog::run`, `ddlog_run`, and Go/Java language
+  bindings based on `ddlog_run`.  This optional callback, invoked by DD workers on
+  each update to an output collection complicated the API and was tricky to use
+  correctly.  Most importantly, it is superseded by the `commit_dump_changes` API.
```

Comment:
Removing the update callback from the external api should allow removing [`Relation.change_cb`](https://github.com/vmware/differential-datalog/blob/master/rust/template/differential_datalog/src/program/mod.rs#L237) if I'm not mistaken

Reply:
That callback's still used internally to implement `commit_dump_change`.

Reply:
Darn, I would like to do a rework of that sometime then since I think it's a source of contention that isn't good for perf. If the callback is properly synchronized, we should just be able to store an `Arc<dyn Callback>` directly, which would eliminate the need for a mutex entirely

Reply:
Good point, that's stupid and needs to be fixed.

Reply:
PS. (but not necessarily by you, I should probably clean my own mess)